src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/RecapitiPersonaFisicaTest.java

```diff
@@ -686,7 +686,20 @@ public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneConferma() {
         }
     }
 
-    @Then("Nella pagina I Tuoi Recapiti si controlla che l'indirizzo pec non sia presente")
+
+    @And("Nel pop up elimina indirizzo pec si clicca sul bottone annulla")
+    public void nelPopUpEliminaIndirizzoPecSiCliccaSulBottoneAnnulla() {
+        logger.info("Si clicca sul bottone conferma");
+        if (recapitiDestinatarioPage.waitLoadPopUpElimina().equalsIgnoreCase("Rimuovi PEC")) {
+            recapitiDestinatarioPage.clickSuAnnullaElimina();
+        } else {
+            recapitiDestinatarioPage.clickSuChiudiPopUp();
+            recapitiDestinatarioPage.eliminaNuovaPec();
+            recapitiDestinatarioPage.eliminaPecEsistente();
+        }
+    }
```

Comment:
I think there is already a step or method that should do this kind of action

Reply:
yes there is similar method , but it doesnt verify that pop-up contains "Rimuovi PEC", and doest have a logic to implement if pop-up is not "Rimuovi PEC"